src/open_weather_map/weather.rs

```diff
@@ -14,7 +15,25 @@ pub struct Weather {
     pub timezone: i64,
     pub id: u32,
     pub name: String,
-    pub code: u32,
+    pub cod: u32,
+}
+
+impl fmt::Display for Weather {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let weather_desc = &self.weather[0].description;
```

Comment:
maybe you can check if weather array is not empty

Reply:
Done in [8e9feff](https://github.com/pxp9/weather_bot_rust/pull/17/commits/8e9feff9b546089bf98bdedf9ba69874ff2cf740)